examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino

```diff
@@ -127,25 +132,48 @@ static int mqtt_client_init_function(
   mqtt_broker_uri_span = az_span_copy(mqtt_broker_uri_span, mqtt_client_config->address);
   az_span_copy_u8(mqtt_broker_uri_span, null_terminator);
 
-  mqtt_config.uri = mqtt_broker_uri;
-  mqtt_config.port = mqtt_client_config->port;
-  mqtt_config.client_id = (const char*)az_span_ptr(mqtt_client_config->client_id);
-  mqtt_config.username = (const char*)az_span_ptr(mqtt_client_config->username);
-
-#ifdef IOT_CONFIG_USE_X509_CERT
-  LogInfo("MQTT client using X509 Certificate authentication");
-  mqtt_config.client_cert_pem = IOT_CONFIG_DEVICE_CERT;
-  mqtt_config.client_key_pem = IOT_CONFIG_DEVICE_CERT_PRIVATE_KEY;
-#else // Using SAS key
-  mqtt_config.password = (const char*)az_span_ptr(mqtt_client_config->password);
-#endif
-
-  mqtt_config.keepalive = 30;
-  mqtt_config.disable_clean_session = 0;
-  mqtt_config.disable_auto_reconnect = false;
-  mqtt_config.event_handle = esp_mqtt_event_handler;
-  mqtt_config.user_context = NULL;
-  mqtt_config.cert_pem = (const char*)ca_pem;
+#if defined(ESP_ARDUINO_VERSION_MAJOR) && ESP_ARDUINO_VERSION_MAJOR >= 3
+    mqtt_config.broker.address.uri = mqtt_broker_uri;
+    mqtt_config.broker.address.port = mqtt_client_config->port;
+    mqtt_config.credentials.client_id = (const char*)az_span_ptr(mqtt_client_config->client_id);
+    mqtt_config.credentials.username = (const char*)az_span_ptr(mqtt_client_config->username);
+
+  #ifdef IOT_CONFIG_USE_X509_CERT
```

Comment:
Nit: Do nested `#ifdef`s need this spacing?

Reply:
No they do not, I just did that to improve readability.